SingularityExecutor/src/main/java/com/hubspot/singularity/executor/task/SingularityExecutorTaskProcessCallable.java

```diff
@@ -39,4 +58,27 @@ public String toString() {
     return "SingularityExecutorTaskProcessCallable [task=" + task + "]";
   }
 
+  private void runHealthCheck() {
+    Optional<HealthcheckOptions> maybeOptions = task.getTaskDefinition().getHealthCheckOptions();
+
+    Optional<String> expectedHealthCheckResultFilePath = task.getTaskDefinition().getHealthCheckResultFilePath();
+    if (maybeOptions.isPresent() && expectedHealthCheckResultFilePath.isPresent()) {
+      try {
+        Integer healthcheckMaxRetries = maybeOptions.get().getMaxRetries().or(configuration.getHealthcheckMaxRetries());
+
+        Retryer<String> retryer = RetryerBuilder.<String>newBuilder()
+            .retryIfResult(path -> !new File(path).exists())
+            .withWaitStrategy(WaitStrategies.fixedWait(1L, TimeUnit.SECONDS))
+            .withStopStrategy(StopStrategies.stopAfterAttempt(healthcheckMaxRetries))
+            .build();
+
+        retryer.call(() -> expectedHealthCheckResultFilePath.get());
```

Comment:
would it make more sense for the exists to be here, and the retryer to a have a retryIf predicate instead (i.e. retry if false)?